qemu/tests/win_driver_install_by_installer.py

```diff
@@ -0,0 +1,419 @@
+        vm.send_key('meta_l-d')
+        time.sleep(30)
+        install_driver_by_installer(session)
+        error_context.context("Check if virtio-win-guest-too.exe "
+                              "is signed by redhat", logging.info)
+        status = session.cmd_status(params["signed_check_cmd"])
+        if status != 0:
+            test.fail('Installer not signed by redhat.')
+        if filter_guest_above_win10(session):
+            error_context.context("Check if QEMU FWCfg Device is installed.",
+                                  logging.info)
+            device_name = "QEMU FWCfg Device"
+            chk_cmd = params["vio_driver_chk_cmd"] % device_name
+            status = session.cmd_status(chk_cmd)
+            if status != 0:
+                test.fail("QEMU FWCfg Device not installed")
+
+    if params.get("run_update_test", "no") == "yes":
+        if params.get("update_from_previous_installer", "no") == "yes":
+            error_context.context("install drivers from previous installer",
+                                  logging.info)
+            vm.send_key('meta_l-d')
+            time.sleep(30)
+            install_driver_by_installer(session)
+        else:
+            for driver_name, device_name, device_hwid in zip(driver_name_list,
+                                                             device_name_list,
+                                                             device_hwid_list):
+                # downgrade iso have no viofs yet
+                if driver_name == "viofs":
+                    continue
+                install_driver_by_virtio_media(session, driver_name,  device_hwid)
+            error_context.context("Install 'qemu-guest-agent' package in guest.",
+                                  logging.info)
+            s_inst, o_inst = session.cmd_status_output(gagent_install_cmd)
+            if s_inst != 0:
+                test.fail("qemu-guest-agent install failed,"
+                          " the detailed info:\n%s." % o_inst)
+
+        error_context.context("Upgrade virtio driver to original",
+                              logging.info)
+        change_virtio_media(params["cdrom_virtio"])
+        vm.send_key('meta_l-d')
+        time.sleep(30)
+        copy_file_to_samepath(session)
+        install_driver_by_installer(session)
+        session = vm.reboot(session)  # workaround as Bug 1847856
+
+    error_context.context("Check if gagent version is correct.",
+                          logging.info)
+    actual_gagent_version = session.cmd_output(gagent_pkg_info_cmd).split()[-2]
+    if actual_gagent_version != expected_gagent_version:
+        test.fail("gagent version is not right, expected is %s but got %s"
+                  % (expected_gagent_version, actual_gagent_version))
+
+    wrong_ver_driver = []
+    not_signed_driver = []
+    for driver_name, device_name in zip(driver_name_list, device_name_list):
+        if driver_name == "qemufwcfg":
+            continue
```

Comment:
qemufwcfg also need to be signed .

Reply:
for qemufwcfg just check driver sign is enough as qemufwcfg driver not change always and is not important